app/src/gecko/java/org/mozilla/focus/web/GeckoViewPermissionsDelegate.kt

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.web
+
+import android.app.Activity
+import android.content.Context
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.util.Log
+
+import org.mozilla.focus.R
+import org.mozilla.focus.permissions.IPermissionsDelegate
+import org.mozilla.gecko.GeckoSession
+
+import java.util.Locale
+
+class GeckoViewPermissionsDelegate(private val context: Context,
+                                   private val geckoSession: GeckoSession) :
+        GeckoSession.PermissionDelegate, IPermissionsDelegate {
+    private var callback: GeckoSession.PermissionDelegate.Callback? = null
+
+    override fun onRequestPermissionsResult(permissions: Array<String>, grantResults: IntArray) {
+        if (callback == null) {
+            return
+        }
+
+        val cb = callback
```

Comment:
You can rewrite this as:

```Kotlin
val cb = callback ?: return
```

With that `cb` should never be null and you can avoid using `!!`